examples/chef/common/stubs.cpp

```diff
@@ -347,22 +360,116 @@ void emberAfWakeOnLanClusterInitCallback(EndpointId endpoint)
 }
 #endif
 
-void ApplicationInit()
+/**
+ * This initializer is for the application having refrigerator on EP1 and child
+ * temperatureControlledCabinet endpoints on EP2 and EP3.
+ */
+void RefrigeratorTemperatureControlledCabinetInit()
 {
-    ChipLogProgress(NotSpecified, "Chef Application Init !!!");
-
-#ifdef MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
-    // set Parent Endpoint and Composition Type for an Endpoint
     EndpointId kRefEndpointId           = 1;
     EndpointId kColdCabinetEndpointId   = 2;
     EndpointId kFreezeCabinetEndpointId = 3;
-    SetTreeCompositionForEndpoint(kRefEndpointId);
-    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
-    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
-    // set TagList
-    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
-    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
-#endif // MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
+    if (DeviceTypes::EndpointHasDeviceType(kRefEndpointId, DeviceTypes::kRefrigeratorDeviceId))
+    {
+        ChipLogDetail(NotSpecified, "Refrigerator device type on EP: %d", kRefEndpointId);
+        SetTreeCompositionForEndpoint(kRefEndpointId);
+
+        // Cold Cabinet
```

Comment:
nit: comment does not seem to augment code: the cold cabinet is in the name and text below.